iml-action-client/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use futures::{future, StreamExt, TryStreamExt};
+use iml_action_runner::ActionType;
+use iml_fs::stream_file_lines;
+use iml_manager_env::{get_action_runner_connect, get_action_runner_socket};
+use iml_wire_types::{Action, ActionId, ActionName, AgentResult, Fqdn};
+use uuid::Uuid;
+
+async fn running_in_docker() -> bool {
+    stream_file_lines("/proc/self/cgroup")
+        .boxed()
+        .try_filter(|l| future::ready(l.split(':').nth(1) == Some("docker")))
+        .try_next()
+        .await
+        .unwrap_or(None)
+        .is_some()
+}
+
+pub async fn invoke_rust_agent(
+    host: impl Into<Fqdn>,
+    command: impl Into<ActionName>,
+    args: serde_json::Value,
+) -> AgentResult {
```

Comment:
We need to return a future oneshot in addition to the result here. The oneshot will then race against the rpc to see who finishes first.

If the oneshot is called, it means we should cancel the RPC in flight.